### PR TITLE
Search Contexts: Fix fetchRepositoriesByNames

### PR DESCRIPTION
**Pull request by gl-srgr**
Issue: https://github.com/sourcegraph/sourcegraph/issues/49978

We will always return an error for static type of search context since we do not pass one of the required parameters `first` or `last`.

Since we are doing a lookup by set of repository names that we want to validate we should get an exhaustive result set and not a truncated result set, so I pass `names.length` for `first`.

## Test plan

sg start; create search contexts with the following static repository count:

- 0
- 1
- 2

## App preview:

- [Web](https://sg-web-garyl-pass-first-fetch-repos-by.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by gl-srgr** [comment]
@limitedmage the issue is in one of the queries defined in backend.ts. I looked at test files in the same directory but I'm very unfamiliar with what to mock in these tests (as opposed to our backend go tests) so I've not proceeded with adding a test file for SearchContextsForm. Let me know if you think we still need it, thanks!

**Review by limitedmage — APPROVED** [review]
Good find! I don't think tests are absolutely necessary here since this is a mistake in how the query was written.

File: client/web/src/enterprise/searchContexts/backend.ts [hunk 1]
```diff
@@ -9,10 +9,11 @@ import { RepositoriesByNamesResult, RepositoriesByNamesVariables } from '../../g
 export function fetchRepositoriesByNames(
     names: string[]
 ): Observable<RepositoriesByNamesResult['repositories']['nodes']> {
+    const first = names.length
     return requestGraphQL<RepositoriesByNamesResult, RepositoriesByNamesVariables>(
         gql`
-            query RepositoriesByNames($names: [String!]!) {
-                repositories(names: $names) {
+            query RepositoriesByNames($names: [String!]!, $first: Int!) {
+                repositories(names: $names, first: $first) {
```
Comment (eseliger) [inline client/web/src/enterprise/searchContexts/backend.ts:16]: this feels like quite an annoying limitation from the API 😢 should we raise that?

**Review by eseliger — COMMENTED** [review]
should we backport this too so it's in 5.0.1 ?

File: client/web/src/enterprise/searchContexts/backend.ts (hunk 1 above)
Comment (gl-srgr) [reply client/web/src/enterprise/searchContexts/backend.ts:16]: I can see how this limitation helps with performance by making the result set have an explicit first or last. There's also an option [`AllowNoLimit`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/graphqlutil/connection_resolver.go?L71) although I'd have to modify the options for connectionResolver which probably means a new query.

Do you think that AllowNoLimit is insufficient? If so we can raise this with repo mgmnt

**Pull request merged**